private/pkg/manifest/module_test.go

```diff
@@ -115,58 +115,43 @@ func TestInvalidNewDigestFromBlobHash(t *testing.T) {
 
 func TestNewBlobSet(t *testing.T) {
 	t.Parallel()
-	var blobs []manifest.Blob
-	for i := 0; i < 10; i++ {
-		content := fmt.Sprintf("some content %d", i)
-		digest := mustDigestShake256(t, []byte(content))
-		blob, err := manifest.NewMemoryBlob(
-			*digest, []byte(content),
-			manifest.MemoryBlobWithHashValidation(),
-		)
-		require.NoError(t, err)
-		blobs = append(blobs, blob)
-	}
-
-	t.Run("Valid", func(t *testing.T) {
-		t.Parallel()
-		blobSet, err := manifest.NewBlobSet(
-			context.Background(),
-			blobs,
-			manifest.BlobSetWithContentValidation(),
-		)
-		require.NoError(t, err)
-		assert.NotNil(t, blobSet)
-	})
+	blobs := newBlobsArray(t)
+	blobSet, err := manifest.NewBlobSet(
+		context.Background(),
+		blobs,
+		manifest.BlobSetWithContentValidation(),
+	)
+	require.NoError(t, err)
+	assert.NotNil(t, blobSet)
+}
 
-	t.Run("BlobSetWithContentValidation", func(t *testing.T) {
-		t.Parallel()
-		t.Run("DuplicatedValidBlobs", func(t *testing.T) {
-			t.Parallel()
-			duplicatedBlobs := append(blobs, blobs[0])
-			blobSet, err := manifest.NewBlobSet(
-				context.Background(),
-				duplicatedBlobs,
-				manifest.BlobSetWithContentValidation(),
-			)
-			require.NoError(t, err)
-			assert.NotNil(t, blobSet)
-		})
+func TestNewBlobValidDuplicates(t *testing.T) {
+	t.Parallel()
+	blobs := newBlobsArray(t)
+	duplicatedBlobs := append(blobs, blobs[0])
```

Comment:
nit: could re-use `blobs` as name now, whenever you see `x = append(<not x>, ...)` it is more potential overhead than the two separate variable names are helpful imo.

Reply:
you're right, and actually this var is not explicitly needed, let me fix